### ISSUE TITLE
Support int8/int16 for packed_store_active/packed_load_active 

### ISSUE THREAD

**Issue by ColonelPhantom** [ispc/ispc#3095]
Needing to generate variable output can be a common pattern (see e.g. [the FAQ](https://ispc.github.io/faq.html#how-can-a-gang-of-program-instances-generate-variable-amounts-of-output-efficiently)).

I would think that instructions like VPCOMPRESS can make this easier and more efficient, since results can go directly where they should and there is no expensive 'post-processing' step involving expensive scans and memory copies.

For reference, the FAQ code (modified to actually put things in `outLocal`) is:
```c
uniform int func(uniform float outArray[], ...) {
   int numOut = ...;  // figure out how many to be output
   float outLocal[MAX_OUT]; // staging area

   // each program instance in the gang puts its results in
   //  outLocal[0], ..., outLocal[numOut-1]
   for (...) {
      if (condition) {
         outLocal[numOut++] = value;
      }
   }

   int startOffset = exclusive_scan_add(numOut);
   for (int i = 0; i < numOut; ++i)
       outArray[startOffset + i] = outLocal[i];
   return reduce_add(numOut);
}
```

With a `compress_store` intrinsic, one could instead write something like:

```c
uniform int func(uniform float outArray[], ...) {
    uniform int numOut = 0;
    for (...) {
        if (condition) {
            numOut += compress_store(&outArray[numOut], value);
        }
    }
    return numOut;
}
```

assuming that `compress_store` returns the number of written elements, and I didn't get any pointer semantics wrong :)

This also changes the order in which elements are written, but I would consider the compress approach to have a more 'logical' result, since something like `foreach (i = 0 ... n) { compress_store(...); }` would also put `i` into the array in order.

I feel like in-register compress would be kind of weird in ISPC, since it would also need to modify the mask and as such mess up the 'scalar illusion', but compress-store can be perfectly explained in ISPC semantics (with `foreach_active`) and does not affect the mask in any way.

**Comment by ColonelPhantom** [comment]
Actually, it seems like it's been there since the beginning, under the name `packed_store_active` (and `packed_load_active`). It also uses the VPCOMPRESS intrinsics.

However, it does seem these are limited to int32 and int64. I think it would be nice if all datatypes were supported.

**Comment by dbabokin** [comment]
Yep, we have these library function, check out https://ispc.github.io/ispc.html#packed-load-and-store-operations for more details. Note that we have also `packed_store_active2()` version, that is a faster version, see the documentation for details.

For other data types, we should probably add them, this makes sense for completeness.
